src/Sound.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include <SFML/Audio.hpp>
+#include <string>
+
+#include "Files.hpp"
+
+class Sound {
+ private:
+  // sound data is not stored directly in sf::Sound, it is stored in sf::SoundBuffer
+  sf::SoundBuffer mBuffer;
+
+  // load Sound from file (it can be loaded from memory, stream, array of samples, which are 16 bit
+  // integers with sign) MP3 is not supproted due to licensing
+  std::string mPath;
+
+  // to play a sound, sf::Sound instance is needed.
+  sf::Sound mSound;
+
+
+ public:
+  Sound();
+  Sound(const Sound& sound);
```

Comment:
You don't have any "special" members that require special handling of the memory (like pointers, smart pointer, references...). In such case you don't need to provide neither the copy constructor nor the assignment operator

Reply:
I need to provide copy constructor if i pass objects by const reference - i need to set sf::SoundBuffer.